### PR TITLE
📦 NEW: add react-query  for data-management 

### PR DESCRIPTION
**Pull request by jnebab**
1. added react-query and config for data management. 
2. implemented mocked data-fetching to use react-query
3. added a separate interface file for types
4. fixed directory mapping error on homepage
5. fixed import error in project info page
6. removed type checking plugin for ts lint config
7. added Seo in project info page

### CODE REVIEW

**Comment by kquirapas** [comment]
Hindi ko malagyan ng suggestion kasi wala sa diff hunk

Pero `line 47-48` consider adding a "key" attribute to the list rendered project tags

```typescript
        {project.tags.map((tag) => (
          <li className="rounded-sm bg-purple-heart px-2 py-1 lowercase text-white">
            {tag}
          </li>
        ))}
```

Do this
```typescript
        {project.tags.map((tag, idx) => (
          <li key={idx} className="rounded-sm bg-purple-heart px-2 py-1 lowercase text-white">
            {tag}
          </li>
        ))}
```

Can you apply this change? @jnebab

**Comment by jnebab** [comment]
> Hindi ko malagyan ng suggestion kasi wala sa diff hunk
> 
> Pero `line 47-48` of `Card.tsx` consider adding a "key" attribute to the list rendered project tags
> 
> ```ts
>         {project.tags.map((tag) => (
>           <li className="rounded-sm bg-purple-heart px-2 py-1 lowercase text-white">
>             {tag}
>           </li>
>         ))}
> ```
> 
> Do this
> 
> ```ts
>         {project.tags.map((tag, idx) => (
>           <li key={idx} className="rounded-sm bg-purple-heart px-2 py-1 lowercase text-white">
>             {tag}
>           </li>
>         ))}
> ```
> 
> Can you apply this change? @jnebab

done na sir

**Comment by kquirapas** [comment]
Pwede na isabay dito yung attachment nung true api. @warengonzaga when are you planning to deploy? or deploy ako ng temporary?

**Comment by john-revzoom** [comment]
> Pwede na isabay dito yung attachment nung true api. @warengonzaga when are you planning to deploy? or deploy ako ng temporary?

that might limit other contributors though. If we wait for the real api, then I probably won't be able to do it until next weekend. :D

**Comment by kquirapas** [comment]
Makes sense

**Comment by warengonzaga** [comment]
> > Pwede na isabay dito yung attachment nung true api. @warengonzaga when are you planning to deploy? or deploy ako ng temporary?
> 
> that might limit other devs to contribute though. ayuko naman ariin lahat ng gagawin hehe

Indeed, we must follow the milestone in the design docs. I'll merge this with mock data only. Let's add them on the next merge.

File: src/mock/index.ts [hunk 1]
```diff
@@ -1,26 +1,6 @@
...
-      id: string;
-      url: string;
-    };
-  };
-}
+import { Project } from "@/interface";
 
-export const directory: Array<Project> = [
+export const directory: Project[] = [
   {
```
Comment (warengonzaga) [inline src/mock/index.ts:4]: Why is there still `Project` in the code base?

File: src/interface/index.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * otherwise, use mixed type arrays (i.e, Array<string, Project, boolean>)
+ * https://mattferderer.com/difference-between-array-types-in-typescript
+ */
+
+export interface SocialMedia {
+  id: string;
+  url: string;
+}
+
+export interface Project {
```
Comment (warengonzaga) [inline src/interface/index.ts:17]: Replace this with `Directory` instead.

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -1,8 +1,19 @@
 import type { AppProps } from "next/app";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import "@/styles/globals.css";
 
+/**
+ * Create a client
+ * https://tanstack.com/query/v4/docs/quick-start
+ */
```
Comment (warengonzaga) [inline src/pages/_app.tsx:8]: Is this necessary here?

File: src/pages/directory/[id].tsx [hunk 4]
```diff
@@ -1,12 +1,17 @@
-import { Project as ProjectType, directory } from "@/mock";
+/* eslint-disable @typescript-eslint/require-await */
+import { GetStaticPropsContext } from "next";
+import Seo from "@/components/Seo";
+import { Project } from "@/interface";
+import { directory } from "@/mock";
 
 interface ProjectProps {
```
Comment (warengonzaga) [inline src/pages/directory/[id].tsx:7]: I prefer `DirectoryProps`

File: src/pages/directory/[id].tsx [hunk 5]
```diff
@@ -1,12 +1,17 @@
...
-  project: ProjectType;
+  project: Project;
```
Comment (warengonzaga) [inline src/pages/directory/[id].tsx:8]: Change to `Directory` almost all.

File: src/pages/directory/[id].tsx [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
 }
 
-export default function Project({ project }: ProjectProps) {
+export default function ProjectPage({ project }: ProjectProps) {
```
Comment (warengonzaga) [inline src/pages/directory/[id].tsx:11]: `DirectoryPage`

File: src/pages/directory/[id].tsx [hunk 7]
```diff
@@ -1,12 +1,17 @@
...
   return (
     <div>
+      <Seo title={project.name} />
       <h3>{`Project Information page for ${project.name}`}</h3>
       <p>{project.description}</p>
```
Comment (warengonzaga) [inline src/pages/directory/[id].tsx:16]: change to `directory`

File: src/pages/directory/[id].tsx [hunk 8]
```diff
@@ -23,12 +28,18 @@ export async function getStaticPaths() {
   };
 }
 
-export async function getStaticProps({ params }: any) {
-  const id = params.id;
+export async function getStaticProps(context: GetStaticPropsContext) {
+  const id = context.params?.id;
   const project = directory.find((p) => p.id === id);
+  if (!!id && !!project) {
```
Comment (warengonzaga) [inline src/pages/directory/[id].tsx:34]: Anything project to `directory`

**Review by warengonzaga — CHANGES_REQUESTED** [review]
Kindly replace all `projects` or `project` to `directory` or `directories`

**Comment by warengonzaga** [comment]
hey, @jnebab let me know once you are done with the changes, basically, just replace `project` in the code base and replace with `directory` instead. Thank you for your help and contribution!

**Comment by jnebab** [comment]
Done @warengonzaga

File: src/pages/index.tsx [hunk 9]
```diff
@@ -2,13 +2,13 @@ import type { NextPage } from "next";
 import { useState } from "react";
 import Card from "@/components/Card";
 import Seo from "@/components/Seo";
-import { Project } from "@/interface";
-import { useGetDirectoryList } from "@/queries";
+import { Directory } from "@/interface";
+import { useGetDirectories } from "@/queries";
```
Comment (warengonzaga) [inline src/pages/index.tsx:6]: It is good to be `useGetDirectoryList`

File: src/pages/index.tsx [hunk 10]
```diff
@@ -2,13 +2,13 @@ import type { NextPage } from "next";
...
 
 const Home: NextPage = () => {
   const [page, setPage] = useState<number>(1);
   const [size, setSize] = useState<number>(10);
-  const { data: directory } = useGetDirectoryList({
+  const { data: directories } = useGetDirectories({
```
Comment (warengonzaga) [inline src/pages/index.tsx:11]: Same here good to be `useGetDirectoryList`

File: src/queries/index.ts [hunk 11]
```diff
@@ -1,23 +1,18 @@
 import { useQuery } from "@tanstack/react-query";
 import { DirectoryQuery } from "@/interface";
-import { directory } from "@/mock";
+import { directories } from "@/mock";
 
 //TODO: make this async await for actual data fetching
-function getDirectoryList() {
+function getDirectories() {
```
Comment (warengonzaga) [inline src/queries/index.ts:6]: if it is referring to a list we can use `directorylist` instead apply it to other.

**Review by warengonzaga — CHANGES_REQUESTED** [review]
Look good to me, just a few changes are needed. Thank you for your contribution @jnebab 💪

File: src/queries/index.ts (hunk 11 above)
Comment (jnebab) [reply src/queries/index.ts:6]: I want to stick with two terms only, directory and directories. using directoryList as well as directories might get confusing later on when the codebase becomes larger.

list - directories
item - directory

Comment (warengonzaga) [reply src/queries/index.ts:6]: I see make sense, all good to me.

**Pull request merged**